### PR TITLE
feat: Use `constructor(…)` operation instead of extended attribute

### LINKED ISSUE #131: Align with new WebIDL constructor syntax
Recently, the WebIDL spec replaced the `[Constructor]` extended attribute by a `constructor` operation ([WebIDL spec PR](https://github.com/heycam/webidl/pull/700)). We are currently in the process of updating all the depending specification to use this new syntax ([HTML spec PR](https://github.com/whatwg/html/pull/4915), [DOM spec PR](https://github.com/whatwg/dom/pull/786/files)).

### PR DESCRIPTION
**Pull request by ExE-Boss**
Resolves #131

---

review?(@pmdartus, @domenic)

### CODE REVIEW

File: lib/overloads.js [hunk 1]
```diff
@@ -2,20 +2,20 @@
 
 const { areDistinguishable, sameType } = require("./types");
 
+/**
+ * @param {string} type
+ * @param {*} A
+ * @param {import("./constructs/interface")} I
+ * @return {import("webidl2").OperationMemberType[] | import("webidl2").ConstructorMemberType[]}
+ */
```
Comment (domenic) [inline lib/overloads.js]: Please don't add this to your PRs; it makes them hard to review.

Comment (ExE-Boss) [reply lib/overloads.js]: In&nbsp;what&nbsp;way does&nbsp;**JSDoc** make&nbsp;code harder&nbsp;to&nbsp;review?

Comment (domenic) [reply lib/overloads.js]: It's a bunch of meaningless noise for humans. I understand it makes some software happier but this is unreadable.

File: lib/constructs/interface.js [hunk 2]
```diff
@@ -76,6 +76,8 @@ class Interface {
     this.implemented = [];
     this.included = [];
 
+    /** @type {import("webidl2").ConstructorMemberType[]} */
```
Comment (domenic) [inline lib/constructs/interface.js]: Also remove here

File: lib/overloads.js (hunk 1 above)
Comment (ExE-Boss) [reply lib/overloads.js]: It’s&nbsp;meaningful to&nbsp;humans&nbsp;too, [as&nbsp;it&nbsp;describes what&nbsp;the&nbsp;function&nbsp;does](https://jsdoc.app/about-getting-started.html).

**Comment by TimothyGu** [comment]
@ExE-Boss Would you be willing to help migrate jsdom to use this new syntax?

**Comment by ExE-Boss** [comment]
@TimothyGu Done&nbsp;in:&nbsp;<https://github.com/jsdom/jsdom/pull/2765>.

**Pull request merged**